### DISCUSSION TITLE
NordVPN with Synology DSM7 not able to start the container

### DISCUSSION THREAD

**Discussion by devzaak** [haugene/docker-transmission-openvpn#2190] [Q&A] [answered]
Hi, I went through the pinned issue related to this combination of issue, but can't seem to get it to work.
Interestingly enough, using Portainer fix I was able to get it working and then yesterday it crashed and is not able to get back into running state.

With privileged mode the error message on start is:
`Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/nordvpn/default.ovpn:1: html (2.5.2)`

without privileged mode or NET_ADMIN error is the old:
`Options error: In [CMD-LINE]:1: Error opening configuration file: /etc/openvpn/nordvpn/default.ovpn`

I know it has been a while since this topic was active but I was wondering if there is any new insights on how to solve this, other than changing VPN provider/NAS etc.

Thank you

**Comment by edgd1er** [comment, marked as answer]
the downloaded file (default.ovpn) is an html error page not a server configuration's file. This is due to either nordvpn's api being unavailable or the throttle limit of the API. (too many container restart)
the dev image has a cache feature to mininize query to the api (+other enhancements)

**Reply by devzaak** [reply]
Ok, what does that mean in practical terms?
How do I get around this?
Should I download the openVPN file myself or use one of the ones in container already somehow?

**Reply by edgd1er** [reply]
It meens that nordvpn won't answer to your requests to the api because the container has been hammering the API.

three workarounds: 
- retry with dev tag container as the cache feature may help not triggering throttle limits.
- stop the container for a while to be forgotten by nordvpn. I have no idea of the delay.
- use dev tag image,  use the new var: NORDVPN_SERVER. usage is explained here: https://github.com/haugene/docker-transmission-openvpn/blob/dev/docs/provider-specific.md

 github pages (doc) are updated when master is updated, hence corresponding to the latest image.

**Reply by devzaak** [reply]
Great, thanks for the comprehensive instructions! I will try them out

**Reply by devzaak** [reply]
Interestingly enough, now with server set it manages to get the openVpn config, but then fails with:
```
Starting container with revision: 20877f1b168b6ff27fc58aeef40756572c562d47
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
INFO: OVPN: Checking curl installation
INFO: OVPN: DNS resolution ok
INFO: OVPN: ok, configurations download site reachable
INFO: OVPN: Removing existing configs in /etc/openvpn/nordvpn
INFO: OVPN: server : it186.nordvpn.com, load: 9
INFO: OVPN: Downloading config: it186.nordvpn.com.ovpn
INFO: OVPN: Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/it186.nordvpn.com.udp.ovpn
Starting OpenVPN using config it186.nordvpn.com.ovpn
Modifying /etc/openvpn/nordvpn/it186.nordvpn.com.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
/etc/openvpn/start.sh: line 283: OPENVPN_OPTS: unbound variable
```
Having looked at the docs, it does not say what this `OPENVPN_OPTS` should be set in this scenario  here: [Docs](https://github.com/haugene/docker-transmission-openvpn/blob/dev/docs/config-options.md) but the docs point only to open vpn documentation.

**Reply by edgd1er** [reply]
well, that's interesting. the start.sh has no -o options (requiring var to be defined).
as a temp fix, you may defined an empty OPENVPN_OPTS in your docker start args.

**Reply by devzaak** [reply]
Looks like that did it, it is starting now after adding `OPENVPN_OPTS` to the env vars with no value
Interesting setup. Thanks for the help!

I think this can be for now marked as answered
